src/github/github.api.ts

```diff
@@ -24,158 +23,28 @@ export class GithubApi {
-        result.user.contributionsCollection.issueContributionsByRepository,
-      );
-    const totalIssueContributions =
-      this.totalContributionsFromRepositories(issueContributions);
-
-    // Pull Request
-    const pullRequestContributions: ContributionsByRepository[] =
-      this.contributionsFromOpenSourceRepositories(
-        result.user.contributionsCollection
-          .pullRequestContributionsByRepository,
-      );
-    const totalPullRequestContributions =
-      this.totalContributionsFromRepositories(pullRequestContributions);
-
-    return {
-      id: result.user.id,
-      username: result.user.login,
-      name: result.user.name,
-      avatarUrl: result.user.avatarUrl,
-      totalContributions:
-        totalRepositoriesContributions +
-        totalCommitContributions +
-        totalIssueContributions +
-        totalPullRequestContributions,
-    };
-  }
+    const query = this.metrics.reduce(
+      (accumulator, currentValue) =>
+        `${accumulator} ${currentValue.buildQuery(contributorUsername)}`,
+      '',
+    );
 
-  /**
-   * Return contributions from open source repositories.
-   */
-  contributionsFromOpenSourceRepositories(
-    contributionsByRepository: ContributionsByRepository[],
-  ): ContributionsByRepository[] {
-    const openSourceLicenses = ['mit', 'apache-2.0', 'gpl-3.0', 'agpl-3.0'];
-    return contributionsByRepository.filter(
-      (contributionByRepository) =>
-        !contributionByRepository.repository.isFork &&
-        !contributionByRepository.repository.isPrivate &&
-        openSourceLicenses.includes(
-          contributionByRepository.repository.licenseInfo?.key,
-        ),
+    const result = await graphQLClient.request(
+      gql`
+        query {
+          ${query}
+        }
+      `,
     );
-  }
 
-  /**
-   * Return created open source repositories.
-   */
-  createdOpenSourceRepositories(
-    createdRepositories: CreatedRepositoryContribution[],
-  ): CreatedRepositoryContribution[] {
-    const openSourceLicenses = ['mit', 'apache-2.0', 'gpl-3.0', 'agpl-3.0'];
-    return createdRepositories.filter(
-      (createdRepository) =>
-        !createdRepository.repository.isFork &&
-        !createdRepository.repository.isPrivate &&
-        openSourceLicenses.includes(
-          createdRepository.repository.licenseInfo?.key,
-        ),
+    const data = this.metrics.reduce(
+      (accumulator, currentValue) => ({
+        ...accumulator,
+        ...currentValue.parseResult(result),
+      }),
+      {},
     );
-  }
 
-  /**
-   * Return total contributions.
-   */
-  totalContributionsFromRepositories(
-    contributionsByRepository: ContributionsByRepository[],
-  ) {
-    let totalContributions = 0;
-    contributionsByRepository.forEach((contributionByRepository) => {
-      totalContributions =
-        totalContributions + contributionByRepository.contributions.totalCount;
-    });
-    return totalContributions;
+    return data;
```

Comment:
```suggestion
    return data as User;
```

If typescript cannot guess the type of "data" because of the reduce, you can infer it this way, it is cleaner than returning any as a type.